docs/basic-tutorial/basic-features(todolist)/11-add-items/add-items-kr.md

```diff
@@ -0,0 +1,172 @@
+
+```js
+const todoInputHandler = newTodo => {
+    setNewTodoItem(newTodo);
+  };
+
+...
+    <View style={styles.inputContainer}>
+      <TextInput
+        style={styles.input}
+        placeholder="Add an item!"
+        placeholderTextColor={'#999'}
+        onChangeText={todoInputHandler}
+        value={newTodoItem}
+        autoCorrect={false}
+      />
+...
+```
+
+TextInput 컴포넌트의 속성으로 `onChangeText`을 추가하였고 그 안에 `todoInputHandler`를 넣어줍니다. 또한 `value` 속성에는 `newTodoItem`을 넣어줍니다. 이제 입력창에 텍스트를 입력하면 실시간으로 입력한 텍스트 값의 상태가 업데이트 되며 `newTodoItem`에는 텍스트 값의 최신 상태가 담기게 됩니다.
+
+다음으로 아이템을 추가해주는 핸들러를 만들겠습니다.
+
+```js
+const addTodoHandler = () => {
+    onAddTodo(newTodoItem);
+    setNewTodoItem('');
+  };
+
+...
+
+      <View style={styles.button}>
+        <Button title={'ADD'} onPress={addTodoHandler} />
+      </View>
+...
+```
+
+`addTodoHandler` 함수 안에는 `onAddTodo`와`setNewTodoItem`가 있습니다. `onAddTodo` 함수는 사용자가 입력한 텍스트 값을 전달 받아 목록에 추가하고, `setNewTodoItem`함수는 입력창을 공백으로 초기화 시키는 역할을 합니다.
+
+Button 컴포넌트에서 기본적으로 제공하는 `onPress` 이벤트에 `addTodoHandler()` 핸들러를 넣어줍니다. 이제 사용자가 ADD 버튼을 누르면 아이템이 추가됩니다.
+
+### TodoList
+
+추가한 아이템을 리스트에 출력하기 위해 TodoList 컴포넌트와 TodoItem 컴포넌트에 아이템을 전달합니다.
+
+app.js 파일에서 `todos`를 TodoList 컴포넌트에 전달합니다.
+
+```js
+<TodoList todos={todos} />
+```
+
+`todos`는 할 일 목록의 객체가 담긴 배열입니다. 따라서 TodoList컴포넌트에서 TodoListItem 컴포넌트로 전달 할 때에는 배열에 담긴 객체 하나하나를 넘겨줘야합니다.
+
+```js
+const TodoList = ({todos}) => {
+  return (
+    <ScrollView contentContainerStyle={styles.listContainer}>
+      {todos.map(todo => (
+        <TodoListItem {...todo} />
+      ))}
+    </ScrollView>
+  );
+};
+```
+
+`map()` 함수를 이용해서 `todos`에 담긴 아이템을 하나씩 TodoListItem 컴포넌트로 전달합니다. 자바스크립트의 디스트럭처링(destructuring) 문법을 이용하면 TodoListItem 컴포넌트에서 아이템 객체에 담긴 값들을 바로 받을 수 있습니다.
+
+각각의 아이템에는 `textValue`, `id`, `checked`라는 **key**와 그에 해당하는 **value**가 담겨 있습니다.
+
+In TodoListItem component, we can get the values from props. Now let's put the textValue on Text area.
+
+아래와 같이 TodoListItem 컴포넌트에서 TodoList 컴포넌트에서 전달한 값들을 받을 수 있습니다.
+
+`<Text>`컴포넌트에 `textValue` 값을 넣어줍니다.
+
+```js
+const TodoListItem = ({textValue, id, checked}) => {
+  return (
+    <View style={styles.container}>
+      <TouchableOpacity>
+          ...
+      </TouchableOpacity>
+      <Text style={[styles.text, styles.strikeText]}>{textValue}</Text>
+      <TouchableOpacity style={styles.buttonContainer}>
+          ...
+      </TouchableOpacity>
+    </View>
+  );
+};
+```
+
+입력창에 텍스트를 입력하여 ADD 버튼을 눌러 추가하면 리스트에 아이템이 추가됩니다.
+
+![](../images/add_items.png "add_items.png")
+
+그런데 시뮬레이터 하단에 경고창이 뜨는 걸 볼 수 있습니다. 경고 문구는 다음과 같습니다. 
+
+`Each child in a list should have a unique 'key' prop.`
+즉, 리스트에 있는 아이템 객체마다 고유한 key값이 필요하다는 말 입니다.
+
+앞서 새로운 아이템 객체를 생성할 때 랜덤으로 id 값을 만들어준 것을 기억하시나요? id 값을 만들어 준 이유가 바로 여기 있습니다. 리스트에 아이템을 출력할 때에는 각 아이템의 고유 번호를 key 값으로 주어야기 때문입니다. (저희는 간단하게 `Math.random()`으로 id를 만들었지만, 이는 중복될 가능성이 있습니다. 따라서 실제 서비스를 만들 때에는 `uuid`와 같은 패키지 등을 이용하여 중복되지 않는 id를 생성해야합니다.)
```

Comment:
External link for `UUID` package would be better for me.

Reply:
Very good suggestion, I added `uuid` package link to the sentence. Thank you!